docs/arcaflow/getting-started.md

```diff
@@ -1,101 +1,185 @@
+steps.post_wait.enabling-->steps.post_wait.enabling.resolved
+steps.post_wait.enabling-->steps.post_wait.starting
+steps.post_wait.enabling-->steps.post_wait.disabled
+steps.pre_wait.deploy-->steps.pre_wait.starting
+input-->steps.stressng.starting
+input-->steps.pcp.starting
+```
 
-## Step 4: Build the plugin
+[Learn more about running workflows &raquo;](/arcaflow/running/){ .md-button }
 
-Next, let's create a `Dockerfile` and build a container image:
+## Writing Workflows
+As a workflow author, you determine the steps of the workflow, how data will pass between the steps, what input is required from the workflow user, and what output will be returned. It is possible to build very complex workflows with data translations, sub-workflows, parallelisim and serialization, and multiple output paths.
 
-```Dockerfile title="Dockerfile"
-FROM python:alpine
+Let's start with something simple. Our workflow will collect a `nickname` input from the user and will pass that input to an example "Hello world!" step. The workflow will also run a UUID generation step in parallel to the example step, and it will return both the UUID and a greeting.
 
-ADD plugin.py /
-RUN chmod +x /plugin.py && pip install arcaflow_plugin_sdk
+In the first part of the `workflow.yaml` file we will define the workflow compatibility version and the input schema for the workflow. In the input schema, we are expecting only a single input called `nickname` with a type of `string`.
 
-ENTRYPOINT ["/plugin.py"]
-CMD []
+```yaml title="workflow.yaml (excerpt)"
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        nickname:   #<<== Input key name
+          display:
+            description: Just a name
+            name: Name
+          required: true
+          type:
+            type_id: string   #<<== Input value type
+...
 ```
-You can now build the plugin container.
 
-=== "Docker"
-    ```
-    docker build -t example-plugin .
-    ```
-
-=== "Podman"
-    ```
-    podman build -t example-plugin .
-    ```
-
-!!! tip
-    Further reading: [Packaging plugins](plugins/packaging.md)
+Next we will define the steps of the workflow. The steps are to be deployed as container images using the `src` tags for the image files. The `arcaflow-plugin-utilities` plugin has multiple steps available, so we indicate with the `step: uuid` parameter which step we want to run. The `arcaflow-plugin-example` plugin has only one step, so the `step` parameter is not required. The `uuidgen` step requires no input, so we pass a blank object `{}` to it. The `example` plugin requires an input structure of `name` with `_type` and `nick` sub-parameters. We statically define `_type: nickname` as part of the step, and then we use the [Arcaflow expression language](/arcaflow/workflows/expressions/) to reference the workflow input value for `nickname` as the input to the plugin's `nick` parameter.
 
```

Comment:
The terminology defining the "image" you pass to the container execution platform is somewhat tangled. When running `podman build` you pass a `--tag` parameter with the image path, including (optionally) the registry, and the actual "tag" appended to that, so this kind of implies that the entire string is a "tag," though the man pages also call it an "image" and an "imageName". <insert looney toons sound of a shaking head>

None of "image", "image name," nor "image path" seem to be the correct shorthand for the whole string. I'll search some more...

Reply:
As I understand it, container images are stored in a _repository_, where each image can be referenced via a (non-unique) _tag_.  An _image name_ is the combination of the _repository name_ and _tag_.  (There is an implicit assumption that a given repository contains various versions of only one image, although there is no way to enforce this.)  _Repositories_ are hosted in _registries_, and they are conventionally grouped within a _registry_ into _organizations_ or _projects_.  So, a _repository name_ is structured as a path consisting of an _registry_, possibly an _organization_, and an individual repo name, separated by slashes, as in `quay.io/arcalot/arcaflow-plugin-baseimage-python-osbase`.

One side note:  the **local** registry is special.  While locally built images in it are addressable as either `localhost/<repo-name>` or just `<repo-name>` (typically without a project in the path), images that you have pulled from remote registries (or tagged for pushing **_to_** remote registries) are stored with their full image names, even though the registry portion refers to a remote registry.  (This makes sense if you think of the local store as a cache for remote registries.)

Reply:
The description of the `--tag` option in the man page for `podman build` makes things fairly clear:
```
   --tag, -t=imageName
       Specifies the name which is assigned to the resulting image if the build process completes successfully.  If 
       imageName does  not  include  a  registry name, the registry name localhost is prepended to the image name.
```
That option, despite its "spelling" is used to specify an _image name_.  And, although it doesn't explicitly say so, I'm reasonably certain that if you omit the portion after the colon, it will be set to `latest`.

The man page for the `podman tag` command is more explicit:
```
SYNOPSIS
       podman tag image[:tag] [target-name[:tag]...] [options]

       podman image tag image[:tag] [target-name[:tag]...] [options]

DESCRIPTION
       Assigns  a  new image name to an existing image.  A full name refers to the entire image name, including the 
       optional tag after the :.  If there is no tag provided, then Podman defaults to latest for both the image and the 
       target-name.
```
So, the _tag_ is part of the _image name_ which includes a _registry name_.